src/models/local/LocalBaseController.js

```diff
@@ -105,6 +105,17 @@ export default class LocalBaseController {
     }
   }
 
+  getContractSourcePath(contractAlias) {
+    const contractName = this.packageFile.contract(contractAlias)
+    if (contractName) {
+      const contractDataPath = Contracts.getLocalPath(contractName)
+      const { compiler, sourcePath } = fs.parseJson(contractDataPath)
```

Comment:
Note that the resulting `sourcePath` does not resolve any import aliases. This means that if the contract is part of a package (such as OpenZeppelin), the source path will be `openzeppelin-solidity/contracts/MyContract.sol`, instead of the correct `node_modules/openzeppelin-solidity/contracts/MyContract.sol`.

You should be able to distinguish between the two, as local contracts have an absolute path (such as `/home/spalladino/Projects/zeppelinos/cli/contracts/mocks/ImplV1.sol`) and the ones in a node module have a relative path.

I'm ok with leaving this as-is now and handling this case in a future PR.

Reply:
As `truffle-flattener` is using the `truffle-resolve` package to resolve dependencies imports, it will automatically prepend the filesystem directory path (i.e., `/home/spalladino/Projects/zeppelinos/cli/node_modules/`) to the relative path that is read from the `Contract.json` file. This is also the expected behaviour when you flatten contracts that are not inside the project `contracts` folder, for example an OpenZeppelin contract.
You can check that implementation [here](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-resolver/npm.js). Please, let me know if I'm misunderstanding you or if I'm wrong.